pkgs/by-name/cs/csmash/package.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, stdenv
+, SDL
+, SDL_image
+, SDL_mixer
+, fetchpatch
+, fetchurl
+, gtk2
+, libintl
+, pkg-config
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "csmash";
+  version = "0.6.6";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/cannonsmash/csmash-${finalAttrs.version}.tar.gz";
+    hash = "sha256-G+CyfaxGuokblVv48SIVG66m7gWzwpYWoJQjJQz9nT0=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [
+    SDL
+    SDL_image
+    SDL_mixer
+    gtk2
+    libintl
+  ];
+
+  patches = [
+    (fetchpatch {
+      url = "https://ftp5.gwdg.de/pub/linux/gentoo/gentoo-portage/games-simulation/cannonsmash/files/cannonsmash-${finalAttrs.version}-gcc41.patch";
```

Comment:
Please add a comment explaining the patch. And don't use the version as a varaible in the patch URL.